diff -ur a/Android.mk b/Android.mk
--- a/Android.mk	2020-06-27 10:07:20.000000000 +0900
+++ b/Android.mk	2020-06-27 10:08:14.000000000 +0900
@@ -54,14 +54,6 @@
     source/video_common.cc
 
 common_CFLAGS := -Wall -fexceptions
-ifneq ($(LIBYUV_DISABLE_JPEG), "yes")
-LOCAL_SRC_FILES += \
-    source/convert_jpeg.cc      \
-    source/mjpeg_decoder.cc     \
-    source/mjpeg_validate.cc
-common_CFLAGS += -DHAVE_JPEG
-LOCAL_SHARED_LIBRARIES := libjpeg
-endif
 
 LOCAL_CFLAGS += $(common_CFLAGS)
 LOCAL_EXPORT_C_INCLUDES := $(LOCAL_PATH)/include
@@ -75,36 +66,4 @@
 
 include $(CLEAR_VARS)
 
 LOCAL_WHOLE_STATIC_LIBRARIES := libyuv_static
-LOCAL_MODULE := libyuv
-ifneq ($(LIBYUV_DISABLE_JPEG), "yes")
-LOCAL_SHARED_LIBRARIES := libjpeg
-endif
-
-include $(BUILD_SHARED_LIBRARY)
-
-include $(CLEAR_VARS)
-LOCAL_STATIC_LIBRARIES := libyuv_static
-LOCAL_SHARED_LIBRARIES := libjpeg
-LOCAL_MODULE_TAGS := tests
-LOCAL_CPP_EXTENSION := .cc
-LOCAL_C_INCLUDES += $(LOCAL_PATH)/include
-LOCAL_SRC_FILES := \
-    unit_test/basictypes_test.cc  \
-    unit_test/color_test.cc       \
-    unit_test/compare_test.cc     \
-    unit_test/convert_test.cc     \
-    unit_test/cpu_test.cc         \
-    unit_test/cpu_thread_test.cc  \
-    unit_test/math_test.cc        \
-    unit_test/planar_test.cc      \
-    unit_test/rotate_argb_test.cc \
-    unit_test/rotate_test.cc      \
-    unit_test/scale_argb_test.cc  \
-    unit_test/scale_test.cc       \
-    unit_test/scale_uv_test.cc    \
-    unit_test/unit_test.cc        \
-    unit_test/video_common_test.cc
-
-LOCAL_MODULE := libyuv_unittest
-include $(BUILD_NATIVE_TEST)
