diff --git a/source/row_neon64.cc b/source/row_neon64.cc
index ead3e6cdd58322463d9af18f24c896c3e5c81c89..4315c7d0c45cbe45f43ffa2828ad8be7b0daa47d 100644
--- a/source/row_neon64.cc
+++ b/source/row_neon64.cc
@@ -2848,6 +2848,7 @@ void RGBToUV444MatrixRow_NEON(const uint8_t* src_rgb,
         "v27");
 }
 
+#if !defined(LIBYUV_DISABLE_NEON_I8MM)
 void ARGBToUV444MatrixRow_NEON_I8MM(const uint8_t* src_argb,
                                     uint8_t* dst_u,
                                     uint8_t* dst_v,
@@ -2885,6 +2886,7 @@ void ARGBToUV444MatrixRow_NEON_I8MM(const uint8_t* src_argb,
       : "cc", "memory", "v0", "v1", "v2", "v3", "v4", "v5", "v16", "v17",
         "v29");
 }
+#endif // !defined(LIBYUV_DISABLE_NEON_I8MM)
 
 // RGB to BT601 coefficients
 // UB   0.875 coefficient = 112
@@ -3274,6 +3276,7 @@ void ARGB4444ToUVRow_NEON(const uint8_t* src_argb4444,
   );
 }
 
+#if !defined(LIBYUV_DISABLE_NEON_I8MM)
 // Process any of ARGB, ABGR, BGRA, RGBA, by adjusting the ArgbConstants layout.
 static void ARGBToUVMatrixRow_NEON_I8MM_Impl(const uint8_t* src,
                                              int src_stride,
@@ -3353,6 +3356,7 @@ void ARGBToUVMatrixRow_NEON_I8MM(const uint8_t* src_argb,
   ARGBToUVMatrixRow_NEON_I8MM_Impl(src_argb, src_stride_argb, dst_u, dst_v,
                                    width, c);
 }
+#endif // !defined(LIBYUV_DISABLE_NEON_I8MM)
 
 void RGB565ToYRow_NEON(const uint8_t* src_rgb565, uint8_t* dst_y, int width) {
   asm volatile(
@@ -3487,6 +3491,7 @@ void ARGBToYMatrixRow_NEON(const uint8_t* src_argb,
         "v19", "v20", "v21", "v22");
 }
 
+#if !defined(LIBYUV_DISABLE_NEON_DOTPROD)
 void ARGBToYMatrixRow_NEON_DotProd(const uint8_t* src_argb,
                                    uint8_t* dst_y,
                                    int width,
@@ -3564,6 +3569,7 @@ void RGBToYMatrixRow_NEON_DotProd(const uint8_t* src_rgb,
         "v17", "v18", "v19", "v20", "v21", "v22", "v23", "v24", "v25", "v26",
         "v27");
 }
+#endif // !defined(LIBYUV_DISABLE_NEON_DOTPROD)
 
 // RGBA expects first value to be A and ignored, then 3 values to contain RGB.
 void RGBToYMatrixRow_NEON(const uint8_t* src_rgb,
@@ -3952,6 +3958,7 @@ void ARGBGrayRow_NEON(const uint8_t* src_argb, uint8_t* dst_argb, int width) {
       : "cc", "memory", "v0", "v1", "v2", "v3", "v4", "v24", "v25", "v26");
 }
 
+#if !defined(LIBYUV_DISABLE_NEON_DOTPROD)
 static const uvec8 kARGBGrayRowCoeffs = {29, 150, 77, 0};
 static const uvec8 kARGBGrayRowIndices = {0, 0, 0, 19, 2, 2, 2, 23,
                                           4, 4, 4, 27, 6, 6, 6, 31};
@@ -3983,6 +3990,7 @@ void ARGBGrayRow_NEON_DotProd(const uint8_t* src_argb,
         [indices] "r"(&kARGBGrayRowIndices)  // %[indices]
       : "cc", "memory", "v0", "v1", "v2", "v3", "v24", "v25");
 }
+#endif // !defined(LIBYUV_DISABLE_NEON_DOTPROD)
 
 // Convert 8 ARGB pixels (32 bytes) to 8 Sepia ARGB pixels.
 //    b = (r * 35 + g * 68 + b * 17) >> 7
@@ -4025,6 +4033,7 @@ void ARGBSepiaRow_NEON(uint8_t* dst_argb, int width) {
         "v21", "v22", "v24", "v25", "v26", "v28", "v29", "v30");
 }
 
+#if !defined(LIBYUV_DISABLE_NEON_DOTPROD)
 static const uvec8 kARGBSepiaRowCoeffs = {17, 68, 35, 0,  22, 88,
                                           45, 0,  24, 98, 50, 0};
 static const uvec8 kARGBSepiaRowAlphaIndices = {3, 7, 11, 15, 19, 23, 27, 31};
@@ -4065,6 +4074,7 @@ void ARGBSepiaRow_NEON_DotProd(uint8_t* dst_argb, int width) {
       : "cc", "memory", "v0", "v1", "v2", "v3", "v4", "v5", "v6", "v7", "v20",
         "v21", "v22", "v24", "v25", "v26", "v28", "v29", "v30");
 }
+#endif // !defined(LIBYUV_DISABLE_NEON_DOTPROD)
 
 // Tranform 8 ARGB pixels (32 bytes) with color matrix.
 // TODO(fbarchard): Was same as Sepia except matrix is provided.  This function
@@ -4128,6 +4138,7 @@ void ARGBColorMatrixRow_NEON(const uint8_t* src_argb,
         "v17", "v18", "v19", "v22", "v23", "v24", "v25");
 }
 
+#if !defined(LIBYUV_DISABLE_NEON_I8MM)
 void ARGBColorMatrixRow_NEON_I8MM(const uint8_t* src_argb,
                                   uint8_t* dst_argb,
                                   const int8_t* matrix_argb,
@@ -4183,6 +4194,7 @@ void ARGBColorMatrixRow_NEON_I8MM(const uint8_t* src_argb,
       : "cc", "memory", "v0", "v1", "v16", "v17", "v18", "v19", "v20", "v21",
         "v22", "v23", "v31");
 }
+#endif // !defined(LIBYUV_DISABLE_NEON_I8MM)
 
 // Multiply 2 rows of ARGB pixels together, 8 pixels at a time.
 void ARGBMultiplyRow_NEON(const uint8_t* src_argb,
diff --git a/source/compare_neon64.cc b/source/compare_neon64.cc
index 36c5e575c0d566159d4063abe958983cf5fc4d9e..953ee5026b9f2c5d746060ea98d11c43889cb086 100644
--- a/source/compare_neon64.cc
+++ b/source/compare_neon64.cc
@@ -153,6 +153,7 @@ uint32_t HashDjb2_NEON(const uint8_t* src, int count, uint32_t seed) {
   return hash;
 }
 
+#if !defined(LIBYUV_DISABLE_NEON_DOTPROD)
 uint32_t HammingDistance_NEON_DotProd(const uint8_t* src_a,
                                       const uint8_t* src_b,
                                       int count) {
@@ -214,6 +215,7 @@ uint32_t SumSquareError_NEON_DotProd(const uint8_t* src_a,
       : "memory", "cc", "v0", "v1", "v2", "v3", "v4", "v5");
   return sse;
 }
+#endif // !defined(LIBYUV_DISABLE_NEON_DOTPROD)
 
 #endif  // !defined(LIBYUV_DISABLE_NEON) && defined(__aarch64__)
 
diff --git a/include/libyuv/compare_row.h b/include/libyuv/compare_row.h
index a08734e90229d4e63188cd5cdb8f2d8b2afc5f3b..97c1d12cb340d5306128995c287278fb853546db 100644
--- a/include/libyuv/compare_row.h
+++ b/include/libyuv/compare_row.h
@@ -61,8 +61,15 @@ extern "C" {
 #if !defined(LIBYUV_DISABLE_NEON) && defined(__aarch64__)
 #define HAS_HASHDJB2_NEON
 
+#if !defined(LIBYUV_DISABLE_NEON_DOTPROD)
 #define HAS_HAMMINGDISTANCE_NEON_DOTPROD
 #define HAS_SUMSQUAREERROR_NEON_DOTPROD
+#endif // !defined(LIBYUV_DISABLE_NEON_DOTPROD)
+#endif
+
+#if !defined(LIBYUV_DISABLE_MSA) && defined(__mips_msa)
+#define HAS_HAMMINGDISTANCE_MSA
+#define HAS_SUMSQUAREERROR_MSA
 #endif
 
 uint32_t HammingDistance_C(const uint8_t* src_a,
diff --git a/include/libyuv/row.h b/include/libyuv/row.h
index 25c146d63220df7421682453e7810e857da4c059..c5c9b4bcfd562b20d5ee2357df36ce18cdce3f13 100644
--- a/include/libyuv/row.h
+++ b/include/libyuv/row.h
@@ -559,6 +559,7 @@ extern "C" {
 #define HAS_P410TOAR30ROW_NEON
 #define HAS_P410TOARGBROW_NEON
 
+#if !defined(LIBYUV_DISABLE_NEON_DOTPROD)
 #define HAS_ABGRTOYJROW_NEON_DOTPROD
 #define HAS_ABGRTOYROW_NEON_DOTPROD
 #define HAS_ARGBGRAYROW_NEON_DOTPROD
@@ -568,10 +569,15 @@ extern "C" {
 #define HAS_RGBATOYJROW_NEON_DOTPROD
 #define HAS_RGBATOYROW_NEON_DOTPROD
 #define HAS_RGBTOYMATRIXROW_NEON_DOTPROD
+#endif // !defined(LIBYUV_DISABLE_NEON_DOTPROD)
 
+#if !defined(LIBYUV_DISABLE_NEON_I8MM)
+#define HAS_ABGRTOUVJROW_NEON_I8MM
+#define HAS_ABGRTOUVROW_NEON_I8MM
 #define HAS_ARGBCOLORMATRIXROW_NEON_I8MM
 #define HAS_ARGBTOUVMATRIXROW_NEON_I8MM
 #define HAS_ARGBTOUV444MATRIXROW_NEON_I8MM
+#endif // !defined(LIBYUV_DISABLE_NEON_I8MM)
 #endif
 
 // The following are available on AArch64 SVE platforms:
